### PR TITLE
incoming: unserialize using numpy rather than Pandas

### PR DESCRIPTION
**Pull request by jd**

### CODE REVIEW

**Comment by chungg** [comment]
i'm torn. this improves aggregated timeseries serialisation maybe ~15%... but makes bounded timeseries unserialisation 6x slower... :(

File: gnocchi/carbonara.py [hunk 1]
```diff
@@ -331,8 +331,8 @@ def unserialize(cls, data, block_size, back_window):
 
     def serialize(self):
         # NOTE(jd) Use a double delta encoding for timestamps
-        timestamps = numpy.insert(numpy.diff(self.ts.index),
-                                  0, self.first.value)
+        timestamps = numpy.insert(numpy.diff(self.ts.index), 0,
+                                  int(datetime64_to_epoch(self.first) * 10e8))
```
Comment (chungg) [inline gnocchi/carbonara.py]: maybe use self.first.astype('timedelta64[ns]')? so we don't need to convert to timedelta64[s] and then multiply by 10e8

Comment (jd) [inline gnocchi/carbonara.py]: `self.first` is already a `timedelta64[ns]`. It seems to work without conversion indeed, I've simplified.

**Comment by jd** [comment]
@chungg yeah, I know it's slower, but we'll need this patch to move on.
But I'm ok to wait for the big-final-patch that will replace `pandas.Series` with Numpy, if you want, before merging this part.

I'm trying to go by small increments when possible. ;)

**Comment by chungg** [comment]
i'm indifferent. we just need to be aware of slow down. we already have a significant slowdown from previous numpy patch.

let's make the edit in comment for now.

**Comment by jd** [comment]
which edit?

File: gnocchi/carbonara.py (hunk 1 above)
Comment (chungg) [inline gnocchi/carbonara.py]: this one. where you said it works without conversion

Comment (jd) [inline gnocchi/carbonara.py]: Oh I though I updated the PR sorry. Facepalm. Sending it now.

**Comment by jd** [comment]
Pull-request updated, HEAD is now cc46876e0ddafa478c63c9b438ff1a8442747ea7

**Pull request merged**